### ISSUE TITLE
ChipField comes up empty when using SelectArrayInput under ReferenceArrayInput 

### ISSUE THREAD

**Issue by Boredude** [marmelab/react-admin#4565]
Using `ReferenceArrayInput` and `SelectArrayInput`, the selected related resources comes up as empty `ChipField`

![image](https://user-images.githubusercontent.com/7926261/77227435-05c96e00-6b89-11ea-9f25-9c0c0ae2413b.png)

This happens only for the pre-populated related items, 
When choosing from the list however, it works as expected:

![image](https://user-images.githubusercontent.com/7926261/77227460-40cba180-6b89-11ea-978e-0d4175f2bf2c.png)

Code snippet: 

```js
<ReferenceArrayInput reference="permissions" source="permissions" allowEmpty>
    <SelectArrayInput label="Permissions">
        <TextField source="name" />
    </SelectArrayInput>
</ReferenceArrayInput>
```
**Environment**

* React-admin version: 3.2.3
* React version: 16.12.0

**Comment by JulienMattiussi** [comment]
I tried to reproduce your issue in the ra simple example (using PostEdit) but without success

![selectarrayinput](https://user-images.githubusercontent.com/39904906/77300042-b6e91900-6ced-11ea-9e6a-1375f7fdf30f.gif)


```js
//in TagReferenceInput.js
<ReferenceArrayInput {...props} filter={{ published: filter }}>
    <SelectArrayInput>
        <TextField source="name" />
    </SelectArrayInput>
</ReferenceArrayInput>
```
Maybe your implementation has something else involved in the issue.

Or can you provide a codesandbox reproducing this issue ?

**Comment by eperedo** [comment]
Hello, I had the same issue, but in my case the problem was the data type of my ids.

```js
post: {
 "title": "Foo",
 "tag_ids": [1, 2] // ids are NUMBERS
}
```

```js
tags: [
 {
 "id": "1", // ID as string
 }
]
```

🙊

**Comment by JulienMattiussi** [comment]
Thanks for the info @eperedo 
When the data types are not consistent between differents parts of your API, the best solution, is to fix it in the API, or when it's not possible, to manage it with custom data translations in the dataprovider.

@Boredude Does the "String" ids correspond to your case ? And maybemanaging it will fix your issue ?

**Comment by fzaninotto** [comment]
No news for some time, closing.

**Comment by kieransouth** [comment]
I tried using this component with asp.net core identity (editing roles against a user) and the chips were empty but the list underneath was being populated. 

Turns out I was doing .ToString() on each GUID before pulling down from the API which converts them to uppercase, whereas the GUIDs I was pulling from my roles resource had not been .ToString()'d, so it was comparing a lowercase GUID with an uppercase GUID and not finding anything. This works fine when your IDs are integers, but if they're strings, make sure you check!
